moveit_core/planning_scene/test/test_multi_threaded.cpp

```diff
@@ -118,9 +118,12 @@ TEST_P(CollisionDetectorTests, Threaded)
   for (unsigned int i = 0; i < THREADS; ++i)
   {
     moveit::core::RobotState* state = new moveit::core::RobotState(planning_scene_->getRobotModel());
+    const moveit::core::JointModelGroup* g_mim = model->getJointModelGroup("mim_joints");
     collision_detection::CollisionRequest req;
-    state->setToRandomPositions();
+    random_numbers::RandomNumberGenerator rng(void);
+    state.setToRandomPositions(g_min, rng);
```

Comment:
This overloaded version of the function you are using here only sets the position of one `JointModelGroup`. E.g. only one arm of the robot. This changes behavior of the test (the other joints position will be uninitialized memory). You would need to add a new function to `RobotState` that allows to write the following:
```suggestion
    state.setToRandomPositions(rng);
```